plugins/lookup/vmware_moid.py

```diff
@@ -0,0 +1,526 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# Copyright: (c) 2021, Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+DOCUMENTATION = r'''
+lookup: vmware_moid
+short_description: Look up MOID for vSphere objects using vCenter REST API.
+description:
+    - Returns MOID of the object contained in the specified path.
+options:
+    _terms:
+        description: objects to query.
+    vcenter_hostname:
+        description:
+            - The hostname or IP address of the vSphere vCenter.
+        required: true
+        type: str
+    vcenter_password:
+        description:
+            - The vSphere vCenter username.
+        required: true
+        type: str
+    vcenter_rest_log_file:
+        description:
+            - 'You can use this optional parameter to set the location of a log file. '
+            - 'This file will be used to record the HTTP REST interaction. '
+            - 'The file will be stored on the host that run the module. '
+        type: str
+    vcenter_username:
+        description:
+            - The vSphere vCenter username.
+        required: true
+        type: str
+    vcenter_validate_certs:
+        default: true
+        description:
+            - Allows connection when SSL certificates are not valid. Set to C(false) when
+              certificates are not trusted.
+        type: bool
+author:
+    - Alina Buzachis (@alinabuzachis)
+version_added: 1.0.1
+requirements:
+    - python >= 3.6
+    - aiohttp
+'''
+
+
+EXAMPLES = r'''
+# lookup sample
+- name: lookup MOID of the object
+  debug: msg="{{ lookup('vmware_moid', {'vm', '/my_dc/host/my_cluster/esxi1.test/test_vm1'}) }}"
+
+- name: lookup MOID of the object in the path
+  debug: msg="{{ lookup('vmware_moid', {'vm', '/my_dc/host/my_cluster/esxi1.test/'}) }}"
```

Comment:
Add an example that shows the use of login credentials.

Reply:
Not yet addressed.